<?php

namespace App\Containers\{{section-name}}\{{container-name}}\Tests\Unit;

use App\Containers\{{section-name}}\{{container-name}}\Data\Factories\{{model}}Factory;
use App\Containers\{{section-name}}\{{container-name}}\Tasks\Delete{{model}}Task;
use App\Containers\{{section-name}}\{{container-name}}\Tests\UnitTestCase;
use App\Ship\Exceptions\NotFoundException;

/**
 * @group {{_container-name}}
 * @group unit
 */
class {{class-name}} extends UnitTestCase
{
    public function testDelete{{model}}(): void
    {
        ${{_model}} = {{model}}Factory::new()->createOne();

        $result = app(Delete{{model}}Task::class)->run(${{_model}}->id);

        $this->assertEquals(1, $result);
    }

    public function testDelete{{model}}WithInvalidId(): void
    {
        $this->expectException(NotFoundException::class);

        $noneExistingId = 777777;

        app(Delete{{model}}Task::class)->run($noneExistingId);
    }
}
